File: lib/SVM/SVM.js
const spawn = require('child_process').spawn

class SVM{
	constructor(pyfile,modelfile){
		this.pyFile = pyfile
		this.modelFile = modelfile
	}

	setPyFile(pyfile){
		this.pyFile = pyfile
	}

	setModelFile(modelfile){
		this.modelFile = modelfile
	}

	predict(x,dim=7){
		return new Promise((solve)=>{
			const pyProcess = spawn('python', [this.pyFile,this.modelFile,x.toString(),dim])
			pyProcess.stdout.on('data', (result)=>{
				result = (result.toString()).split(',')
				result = result.map((value)=>{
					return parseInt(value)
				})

				solve(result)
			})
		})
	}
}

module.exports = SVM

File: lib/SVM/svm.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import sys
import numpy as np
from sklearn import datasets
from sklearn.externals import joblib

argvs = sys.argv
model = joblib.load(argvs[1])
data = np.array(argvs[2].split(','),dtype=np.float).tolist()
dim = int(argvs[3])
data = [data[x:x+dim] for x in range(0,len(data),dim)]
result = model.predict(data)
result = map(str,result)
sys.stdout.write(','.join(result))